src/coreclr/src/jit/morph.cpp

```diff
@@ -6694,6 +6685,16 @@ bool Compiler::fgCanFastTailCall(GenTreeCall* callee, const char** failReason)
     assert(!callee->AreArgsComplete());
 
     fgInitArgInfo(callee);
+
+    if (!opts.compFastTailCalls)
+    {
+        if (failReason)
+        {
+            *failReason = "Configuration doesn't allow fast tail calls";
+        }
+        return false;
+    }
+
```

Comment:
Can we move it even farther below and use the `reportFastTailCallDecision` lambda? Then it will also go through the fast tail call decisions reporting there is in debug.

Reply:
Makes sense, done.